### PR TITLE
Update testing docs to use MockDeepProxy interface instead of MockProxy interface

### PR DESCRIPTION
**Pull request by amast09**
I had type errors when using the example docs for testing. These updates made the type errors go away.

If there are additional things I need to take care of for an official PR let me know.

Thanks for all your hard work!

### CODE REVIEW

**Comment by molebox** [comment]
Hey @amast09, thanks for the PR! Would you be able to provide a simple reproduction of the errors you were seeing? Also, i cant find the `MockDeepProxy` interface on the `jest-mock-extended` package, it doesnt exist. So im curious to see how this works on your end.

File: content/300-guides/150-testing/100-unit-testing.mdx [hunk 1]
```diff
@@ -84,7 +84,7 @@ The following steps guide you through mocking the Prisma Client using a singleto
 
    ```ts file=singleton.ts
    import { PrismaClient } from '@prisma/client'
-   import { mockDeep, mockReset, MockProxy } from 'jest-mock-extended'
+   import { mockDeep, mockReset, MockDeepProxy } from 'jest-mock-extended'
```
Comment (molebox) [inline content/300-guides/150-testing/100-unit-testing.mdx]: This interface doesnt seem to exist on the `jest-mock-extended` package...

**Review by molebox — COMMENTED** [review]
Hey @amast09, thanks for the PR! Would you be able to provide a simple reproduction of the errors you were seeing? Also, i cant find the MockDeepProxy interface on the jest-mock-extended package, it doesnt exist. So im curious to see how this works on your end.

**Comment by amast09** [comment]
@molebox no problem at all :+1: 

> Would you be able to provide a simple reproduction of the errors you were seeing?

Here is sample code that gives me an error,

```
interface Foo {
  readonly bar: {
    readonly getBaz: () => Promise<number>;
  };
}

describe("Foo tests", () => {
  it("fails to type-check", () => {
    const mockFoo: MockProxy<Foo> = mockDeep<Foo>();
    mockFoo.bar.getBaz.mockResolvedValue(9);
  });
});
```

here is the error it gives me,

`error TS2339: Property 'mockResolvedValue' does not exist on type '() => Promise<number>'`

>  Also, i cant find the MockDeepProxy interface on the jest-mock-extended package

Hmmm, maybe I am using a newer version of the library than was used for these docs, here is the interface in [jest-mock-extended](https://github.com/marchaos/jest-mock-extended/blob/c7b5cba2da72524b7460fbea2297053ac653e8c0/src/Mock.ts#L39).

I am using version `2.0.1` of `jest-mock-extended` which is the latest version.

Let me know if you need anything else, thanks again for your hard work on open source :pray:

**Comment by molebox** [comment]
Hey @amast09,  I updated the version of `jest-mock-extended` to use the same as you. Im still not seeing that type `MockDeepProxy`. I checked the file you linked and it does indeed exist, it was added 3 days ago. Its not however been exported according to [this PR](https://github.com/marchaos/jest-mock-extended/commit/cc29d6d152b8231510a4e687f845b7538ae576e3#diff-a2a171449d862fe29692ce031981047d7ab755ae7f84c707aef80701b3ea0c80) So im wondering how you are able to import it? Are you pulling it in directly from the `node_modules` ? Would you mind sharing a replica of your file where all this happens? Or a repo would be even better.

File: content/300-guides/150-testing/100-unit-testing.mdx (hunk 1 above)
Comment (amast09) [reply content/300-guides/150-testing/100-unit-testing.mdx]: Ah, sorry, this is not the import I was using. I corrected it!

**Comment by amast09** [comment]
@molebox here is a repository that reproduces the issue,

https://github.com/amast09/repro-type-issue

Let me know if this helps or you are still not able to reproduce it.

Thanks again!

**Comment by molebox** [comment]
Thanks for raising this @amast09 and for your examples. Ive [updated the docs with the correct import](https://github.com/prisma/docs/pull/2110) so will close this PR.

**Pull request closed without merging**